cosmos/cache.py

```diff
@@ -400,3 +412,72 @@ def create_cache_profile(version: str, profile_content: str) -> Path:
     profile_yml_path = profile_yml_dir / DEFAULT_PROFILES_FILE_NAME
     profile_yml_path.write_text(profile_content)
     return profile_yml_path
+
+
+def is_cache_package_lockfile_enabled(project_dir: Path) -> bool:
+    package_lockfile = project_dir / PACKAGE_LOCKFILE_YML
+    return package_lockfile.is_file()
+
+
+def _get_sha1_hash(yaml_file: Path) -> str:
+    """Read package-lock.yml file and return sha1_hash"""
+    with open(yaml_file) as file:
+        yaml_content = file.read()
+    data = yaml.safe_load(yaml_content)
+    sha1_hash: str = data.get("sha1_hash", "")
+    return sha1_hash
+
+
+def _get_latest_cached_package_lockfile(project_dir: Path) -> Path | None:
+    """
+    Retrieves the latest cached package-lock.yml for the specified project directory,
+    or creates and caches it if not already cached and hashes match.
+    """
+    cache_identifier = project_dir.name
+    package_lockfile = project_dir / PACKAGE_LOCKFILE_YML
+    cached_package_lockfile = cache_dir / cache_identifier / PACKAGE_LOCKFILE_YML
+
+    try:
+        if cached_package_lockfile.exists() and cached_package_lockfile.is_file():
+            project_sha1_hash = _get_sha1_hash(package_lockfile)
+            cached_sha1_hash = _get_sha1_hash(cached_package_lockfile)
+            if project_sha1_hash == cached_sha1_hash:
```

Comment:
what should happen when the hash does not match?

Reply:
Thanks for the feedback @pankajkoti. I have updated to update the cache in case hash does not match